lib/Runtime/Language/AsmJsModule.cpp

```diff
@@ -1080,6 +1080,11 @@ namespace Js
     AsmJsFunc* AsmJsModuleCompiler::CreateNewFunctionEntry( ParseNode* pnodeFnc )
     {
         PropertyName name = ParserWrapper::FunctionName( pnodeFnc );
+        if ( !name )
```

Comment:
No, this should be invalid asm.js. Returning null here will invalidate the asm.js module and then be parsed as normal javascript